docs/programmers_guide/witness_formal_spec.md

```diff
@@ -17,13 +17,14 @@ In the end, when there are no more instructions left, there MUST be only one ite
 
 ### Nodes
 
-There are multiple types of nodes that can be present on the stack.
-
-- `hashNode`
-- `branchNode`
-- `accountNode`
-- `valueNode`
-- `extensionNode`
+```
+type Node = nil
+          | HashNode (raw_hash:Byte[32])
+          | ValueNode (raw_value:Byte[])
+          | AccountNode (nonce:Uint balance:int storage:Node storage_hash:byte[32] code_hash:byte[32])
```

Comment:
There is no need to distinguish int and unit for the purposes for this semantics, because we don't do signed/unsigned arithmetics with them